### ISSUE TITLE
Add an alternative initial condition for simple harmonic oscillator

### ISSUE THREAD

**Issue by cmp0xff** [kausalflow/hamilflow#35]
In the Lagrangian action of an undamped simple harmonic oscillator (#34), one can easily find the first integral
$$\frac{1}{2} m \dot x^2 + \frac{1}{2} m \omega^2 x^2 \equiv E \eqqcolon \frac{1}{2} m \omega^2 x_0^2\\,,$$
where $E$ is the integral or motion (constant), having the interpretation of energy, and $x_0$ can be interpreted as the maximal displacement.

Integrating this _first_-order differential equation _directly_ gives rise to another constant of motion $t_0$,
$$t-t_0 = \frac{1}{\omega} \arccos \frac{x}{x_0}\\,.$$
In other words, the motion of the system is _directly_ given by the two constant $(x_0, t_0)$, in this framework of Lagrangian mechanics.

I propose we add the initial condition $(x_0, t_0)$, in addition to $(x_0, v_0)$, the latter of which is closer to the _second_-order differential equation of motion, which is less appealing in analytical mechanics.

**Comment by cmp0xff** [comment]
Currently we provide users intermediate variables which are used for analytical or numerical purposes. Maybe we should remove them.
